tests/testutils/src/main/kotlin/io/element/android/tests/testutils/lambda/LambdaRecorder.kt

```diff
@@ -27,6 +27,26 @@ abstract class LambdaRecorder internal constructor(
     }
 }
 
+/**
+ * A recorder that can be used to record the parameters of a suspend lambda invocation.
+ */
+abstract class LambdaSuspendRecorder internal constructor(
```

Comment:
Good call. At first I though it wouldn't be possible since I was getting other errors related to `NoSuchMethodError` but it seems like using `runBlocking` properly does help.

Reply:
https://github.com/element-hq/element-x-android/pull/3594/commits/7202b2df5e5d10f4fc2152428a12588b8f6ae15e

Reply:
Thanks. Let me check why it was necessary to add `return runBlocking { block() }` in all the `Lambda*Recorder` classes. Since `block` does not suspend, I do not see how it helps. I was thinking about adding it on demand at the call site.